src/main/java/org/commcare/util/EncryptionHelper.java

```diff
@@ -0,0 +1,137 @@
+package org.commcare.util;
+
+
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.spec.GCMParameterSpec;
+
+public class EncryptionHelper {
+
+    // these key algorithm constants are to be used only outside of any Keystore scope
+    public static final String CC_KEY_ALGORITHM_AES = "AES";
+    public static final String CC_KEY_ALGORITHM_RSA = "RSA";
+
+    public static final String CC_IN_MEMORY_ENCRYPTION_KEY_ALIAS = "cc-in-memory-encryption-key-alias";
+
+    public enum CryptographicOperation {Encryption, Decryption}
+
+    private IEncryptionKeyProvider encryptionKeyProvider;
+    private boolean useKeyStoreIfAvailable;
+
+    public EncryptionHelper(boolean useKeyStoreIfAvailable) {
+        encryptionKeyProvider = EncryptionKeyServiceProvider.getInstance().serviceImpl(useKeyStoreIfAvailable);
+        this.useKeyStoreIfAvailable = useKeyStoreIfAvailable;
+    }
+
+    public IEncryptionKeyProvider getEncryptionKeyProvider() {
+        return encryptionKeyProvider;
+    }
+
+    public void setEncryptionKeyProvider(IEncryptionKeyProvider newEncryptionKeyProvider) {
+        encryptionKeyProvider = newEncryptionKeyProvider;
+    }
+
+    public void reloadDefaultEncryptionKeyProvider() {
+        encryptionKeyProvider = EncryptionKeyServiceProvider.getInstance().serviceImpl(useKeyStoreIfAvailable);
```

Comment:
This is basically to restore to the default provider after calling the `set` method to change it. And the application is during Unit Tests, in which we don't have the Android env but we need to force the `EncryptionHelper` to use the `KeyStoreEncryptionKeyProvider`. This can actually be deleted as we don't have any other use once the test is done.